src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/aggregators/HeatmapOfflineAggregator.scala

```diff
@@ -0,0 +1,99 @@
+package com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.aggregators
+
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.writer.SqlRowWriter
+import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.{CassandraHeatmapTiles, CassandraPopularPlaces, CassandraTileBucket}
+import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.dto.{ComputedTile, Event, HeatmapTile, PopularPlace}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, SaveMode, SparkSession}
+
+class HeatmapOfflineAggregator(session: SparkSession) extends OfflineAggregator[HeatmapTile] {
+  private val ComputedTilesTable = "computedtiles"
+
+  override def aggregate(events: RDD[Event]): RDD[HeatmapTile] = {
+    val tiles = events.flatMap(event=>{
+      Seq(
+        event,
+        event.copy(externalsourceid = "all"),
+        event.copy(pipelinekey = "all", externalsourceid = "all")
```

Comment:
Does this also need to include `event.copy(pipelinekey = "all")` so that we can query by an arbitrary pipelinekey without also having to restrict the externalsourceid?

Reply:
We're doing this so we can support GQL queries where either `externalsourceid`  or `pipelinekey` are set to `all`